### PR TITLE
document typescript usage

### PR DESCRIPTION
**Pull request by zeke**
I was using the Replicate client in a TypeScript project with an aggressive linter today, and it wasn't immediately obvious to me how to specify the types.

This PR adds a bit of documentation to the README that shows how to import the types ESM-style, and how to use them in a real-world example.

### CODE REVIEW

**Review by superhighfives — APPROVED** [review]
This looks great—thanks for taking the time to write it up, @zeke! 👍

File: README.md [hunk 1]
```diff
@@ -112,6 +112,32 @@ const output = await replicate.run(model, { input });
...
+
+```ts
+import Replicate, { Prediction } from 'replicate';
+
+const replicate = new Replicate();
+const model = "black-forest-labs/flux-schnell";
+const prompt = "a 19th century portrait of a raccoon gentleman wearing a suit";
+const onProgress = (prediction: Prediction) => {
+  console.log({ prediction });
+};
```
Comment (erbridge) [inline README.md:133]:
```suggestion
function onProgress(prediction: Prediction) {
  console.log({ prediction });
}
```

File: README.md [hunk 2]
```diff
@@ -112,6 +112,32 @@ const output = await replicate.run(model, { input });
 > upload the file to your own storage provider
 > and pass a publicly accessible URL.
 
+## TypeScript usage
+
+This library exports TypeScript definitions. You can import them like this:
+
+```ts
+import Replicate, { Prediction } from 'replicate';
```
Comment (erbridge) [inline README.md:120]:
```suggestion
import Replicate, { type Prediction } from 'replicate';
```

File: README.md [hunk 3]
```diff
@@ -112,6 +112,32 @@ const output = await replicate.run(model, { input });
...
+```
+
+Here's an example that uses the `Prediction` type with a custom `onProgress` callback:
+
+```ts
+import Replicate, { Prediction } from 'replicate';
```
Comment (erbridge) [inline README.md:126]:
```suggestion
import Replicate, { type Prediction } from 'replicate';
```

**Comment by zeke** [comment]
Thanks for the fixes, @erbridge 🩷

**Pull request merged**